include/exadg/operators/inverse_mass_operator.h

```diff
@@ -246,122 +332,21 @@ class InverseMassOperator
-  unsigned int quad_index;
-
-  InverseMassParametersHdiv parameters;
-};
-
-/*
- * Inverse mass operator for H(div)-conforming space:
- *
- * This class applies the inverse mass operator by solving the mass system as a global linear system
- * of equations for all degrees of freedom. It is used in case the mass operator is not
- * block-diagonal and can not be inverted element-wise (e.g. H(div)-conforming space).
- */
-template<int dim, int n_components, typename Number>
-class InverseMassOperatorHdiv
-{
-private:
-  typedef dealii::LinearAlgebra::distributed::Vector<Number> VectorType;
-
-public:
-  void
-  initialize(dealii::MatrixFree<dim, Number> const &   matrix_free,
-             dealii::AffineConstraints<Number> const & constraints,
-             InverseMassOperatorDataHdiv const         inverse_mass_operator_data)
-  {
-    // mass operator
-    MassOperatorData<dim, Number> mass_operator_data;
-    mass_operator_data.dof_index  = inverse_mass_operator_data.dof_index;
-    mass_operator_data.quad_index = inverse_mass_operator_data.quad_index;
-    mass_operator.initialize(matrix_free, constraints, mass_operator_data);
-
-    Krylov::SolverDataCG solver_data;
-    solver_data.max_iter             = inverse_mass_operator_data.parameters.solver_data.max_iter;
-    solver_data.solver_tolerance_abs = inverse_mass_operator_data.parameters.solver_data.abs_tol;
-    solver_data.solver_tolerance_rel = inverse_mass_operator_data.parameters.solver_data.rel_tol;
-
-    if(inverse_mass_operator_data.parameters.preconditioner == PreconditionerMass::None)
-    {
-      solver_data.use_preconditioner = false;
-    }
-    else if(inverse_mass_operator_data.parameters.preconditioner == PreconditionerMass::PointJacobi)
-    {
-      preconditioner =
-        std::make_shared<JacobiPreconditioner<MassOperator<dim, n_components, Number>>>(
-          mass_operator, true /* initialize_preconditioner */);
-
-      solver_data.use_preconditioner = true;
-    }
-    else
-    {
-      AssertThrow(false, dealii::ExcMessage("Not implemented."));
-    }
-
-    solver =
-      std::make_shared<Krylov::SolverCG<MassOperator<dim, n_components, Number>,
-                                        PreconditionerBase<Number>,
-                                        VectorType>>(mass_operator, *preconditioner, solver_data);
-  }
-
-  /**
-   * This function applies the inverse mass operator. Note that this function allows identical dst,
-   * src vector, i.e. the function can be called like apply(dst, dst).
-   */
-  unsigned int
-  apply(VectorType & dst, VectorType const & src) const
-  {
-    Assert(solver.get() != 0, dealii::ExcMessage("Mass solver has not been initialized."));
-
-    VectorType temp;
-
-    // Note that the inverse mass operator might be called like inverse_mass.apply(dst, dst),
-    // i.e. with identical destination and source vectors. In this case, we need to make sure
-    // that the result is still correct.
-    if(&dst == &src)
```

Comment:
here we explicitly considered the case `&src == &dst`, but in my tests this is not even needed?